Pre-release activities (#506)
insightsengineering/teal.slice: vignettes/internal_functions_example.Rmd

Reviewer: 1. Separate examples into sections: `FilteredData, `FilteredDataset`, `FilterState`, other.
2. Remove "Example for" from section titles.
3. Rename file to `internal_function_examples.Rmd.
4. Change title to _Examples for internal functions_.

Author: > Separate examples into sections: FilteredData, FilteredDataset, FilterState`, other.

I originally placed these under the suggested section, but I removed them for better navigation through the indexes.
<img width="350" alt="image" src="https://github.com/insightsengineering/teal.slice/assets/6700955/5bb56b2a-b135-4611-99af-20eeb253b1df">

Author: file is too large to navigate via section

Reviewer: What do you mean?

Author: Sorry, I created the section incorrectly and as a result, the examples title didn't show up on the index. I will add the section again.

Author: updated Rmd
<img width="346" alt="image" src="https://github.com/insightsengineering/teal.slice/assets/6700955/af41d697-1ac2-41b7-acc5-6d1a23933a03">

Reviewer: OK, so what's wrong with this?

Author: Nothing, the screenshot is an updated version, I pushed changes for it .

Reviewer: > file is too large to navigate via section

So this is no longer relevant?

Author: yes

---
title: "Interal functions example"
author: "NEST CoreDev"
output: 
  rmarkdown::html_vignette:
    toc: true
vignette: >
  %\VignetteIndexEntry{interal-functions-example}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

## Introduction

> This vignette designed to showcase examples for internal classes, methods, and functions of  teal.slice, specifically designed developers.

Reviewer: ```suggestion
This vignette contains examples for internal classes, methods, and functions of  `teal.slice`. 
For developer use only.
```
